main.go

```diff
@@ -25,7 +32,109 @@ var pongMessage = []byte(`{
 	]
 }`)
 
+type State struct {
+	db *sql.DB
+}
+
+var s *State
+
+func NewState(path string) (*State, error) {
+
+	// connection string
+	psqlconn := path
+
+	db, err := sql.Open("postgres", psqlconn)
+	if err != nil {
+		return nil, err
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println("Connected!")
+
+	s := &State{
+		db: db,
+	}
+	return s, nil
+}
+
+func (m Msg) decodeMsg(field string) interface{} {
+
+	// FORMAT of msg
+	// {
+	// 	"emit": [
+	// 	   "..",
+	// 	   {
+	// 		   "block": {
+	//				number : xxxxx
+	//				hash : xxxxx
+	//				...
+	//			}
+	// 	   }
+	// 	]
+	// }
+
+	decodedMsg := (m.msg["emit"]).([]interface{})[1].(map[string]interface{})["block"].(map[string]interface{})[field]
+	return decodedMsg
+}
+
+type Msg struct {
+	msg map[string]interface{}
+}
+
+func (s *State) WriteBlock(message []byte, table string) {
```

Comment:
Looks good but this function should take a Block struct instead of a raw message it has to decode internally. For example:
```
func (s *State) WriteBlock(block *Block) error {
 ...
}
```
it is easier to test. Luckily, this struct is already done in [this](https://github.com/umbracle/go-web3) library.

Reply:
How to use the web3.Block? Check the test of the library, something like this would work:
```
data := []byte("{"number": "0x1", ...}")
var block web3.Block
if err := json.Unmarshal(data, block); err != nil {
```
You need to figure out how to extract the `block` field from the message.